src/idl_gen_cpp.cpp

```diff
@@ -719,6 +723,71 @@ class CppGenerator : public BaseGenerator {
       }
     }
     code += "  return builder_.Finish();\n}\n\n";
+
+    //Generate a CreateX function with vector types as parameters
+    if (gen_vector_pars) {
+      code += "inline flatbuffers::Offset<" + struct_def.name + "> Create";
+      code += struct_def.name;
```

Comment:
This entire block of code appears copy pasted from the regular codegen for `Create`? Lift out into a function with a bool for `gen_vector_pars`